datasets/modeled/transport/dimension_mappings/make_model_year_mappings.R

```diff
@@ -0,0 +1,84 @@
+
+# for DECARB Task 6: make interpolations/extrapolations for 2024/2026/2028 from 2025/2030 data based on DECARB Task 2 results
+library(tidyverse)
+psgr <- read_csv("C:/Users/ayip/Downloads/decarb2/passenger_energy_decarb_0923_utf8.csv")
+frgt <- read_csv("C:/Users/ayip/Downloads/decarb2/freight_energy_decarb_0923_utf8.csv")
+
+(decarb2_high_ev_load <- psgr %>%
+  filter(Year %in% c(2024,2025,2026,2028,2030,2050),
+         Scenario == "High",
+         str_detect(Fuel, "Electricity")) %>%
+  bind_rows(frgt %>%
+  filter(Year %in% c(2024,2025,2026,2028,2030,2050),
+         Scenario == "High",
+         str_detect(Fuel, "Electricity"))) %>%
+  group_by(Year) %>%
+  summarize(TWh = sum(Quads)*293.0711))
+
+my_to_my <- tibble(from_id = c(2025,2025,2030,2025,2030,2030,2035,2040,2045,2050),
+       to_id =   c(2024,2026,2026,2028,2028,2030,2035,2040,2045,2050),
+       fraction =c(decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2024]/decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2025],
+                   decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2026]/decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2025] * 0.5,
+                   decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2026]/decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2030] * 0.5,
+                   decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2028]/decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2025] * 0.5,
+                   decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2028]/decarb2_high_ev_load$TWh[decarb2_high_ev_load$Year==2030] * 0.5,
+                   1,1,1,1,1))
+
+write_csv(my_to_my, "c:/users/ayip/documents/github/dsgrid-project-DECARB/datasets/modeled/transport/dimension_mappings/model_year_to_model_year.csv")
+
+
+# make counties dimensions and mapping
+library(tidyverse)
+counties <- read_csv("c:/users/ayip/OneDrive - NREL (1)/counties.csv", col_types = "c")
+
+project_counties <- read_csv("C:/Users/ayip/Documents/GitHub/dsgrid-project-DECARB/project/dimensions/counties.csv")
+
+county_to_county <- counties %>% rename(from_id = geography) %>%
+  full_join(project_counties, by = c("from_id" = "id"), keep = TRUE) %>%
+  rename(to_id = id) %>%
+  mutate(to_id = if_else(is.na(state), "", to_id)) %>% #  %in% c("AK","HI")
+  select(-time_zone, -name, -state) %>%
+  filter(!is.na(from_id)) %>%
+  add_row(from_id = "46102", to_id = "46102") %>%
+  arrange(from_id)
+
+NA_either <- county_to_county %>% filter(is.na(from_id) | is.na(to_id) | from_id == "46113" | to_id == "46113")
+NA_either
+
+county_to_county %>%
+  write_csv("C:/Users/ayip/Documents/GitHub/dsgrid-project-DECARB/datasets/modeled/transport/dimension_mappings/county_to_county.csv")
+
+counties %>%
+  rename(id = geography) %>%
+  left_join(project_counties, by = "id") %>%
+  select(id, name, state) %>%
+  add_row(id = "46102", name = "Oglala Lakota", state = "SD") %>%
```

Comment:
Does this mean that you don’t have data for this county? If so, you’ll need to add rows in the dataset with the time columns set to null. That tells dsgrid that the data for those combinations of dimensions is “expected-missing.” As of now, dsgrid will fail registration.

Reply:
Thanks Dan - that's correct that we don't have data for this county.

options: 
1) This county is in the project dimensions. Can't I just leave it out of the dataset dimensions (with an explanation)?

2) If not, per your suggestion, I'll add a parquet file in decarb6-transport-20240207 on eagle that so you can generate one table again.

Reply:
ok, per teams convo, will add parquet

Reply:
That county might exist in your data under its old name: "Oglala Lakota County was renamed in May of 2015, originally it had been named Shannon County", https://oglalalakota.sdcounties.org/

Reply:
understood but we don't have data for either FIPS 46113 or 46102